### ISSUE TITLE
 ERROR  Cannot find module 'react/jsx-runtime' : building "Content" with Solidjs

### ISSUE THREAD

**Issue by doompatrool** [wxt-dev/wxt#463]
Error:
```
WXT 0.16.8                                                                                   10:17:04 PM
ℹ Generating types...                                                                       10:17:06 PM

 ERROR  Cannot find module 'react/jsx-runtime'                                               10:17:06 PM  
Require stack:
- D:\EXTENSION\chrome-nuxt\entrypoints\content\index.jsx

  Require stack:
  - entrypoints\content\index.jsx
  at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
  at Function.resolve (node:internal/modules/cjs/helpers:127:19)
  at _resolve (node_modules\.pnpm\jiti@1.21.0\node_modules\jiti\dist\jiti.js:1:251148)
  at jiti (node_modules\.pnpm\jiti@1.21.0\node_modules\jiti\dist\jiti.js:1:253746)
  at D:/EXTENSION/chrome-nuxt/entrypoints/content/index.jsx:23:26
  at evalModule (node_modules\.pnpm\jiti@1.21.0\node_modules\jiti\dist\jiti.js:1:256443)
  at jiti (node_modules\.pnpm\jiti@1.21.0\node_modules\jiti\dist\jiti.js:1:254371)
  at importEntrypointFile (/D:/EXTENSION/chrome-nuxt/node_modules/.pnpm/wxt@0.16.8/node_modules/wxt/dist/cli.js:2015:23)

```

`App.jsx`
```
import { createSignal } from "solid-js";

export default () => {
  const [count, setCount] = createSignal(1);
  const increment = () => setCount((count) => count + 1);

  return (
    <div>
      <p>This is SOLID. {count}</p>
      <button onClick={increment}>Increment</button>
    </div>
  );
};

```

`Index.jsx`
```

import App from './App.jsx';
import {render } from 'solid-js/web';

export default defineContentScript({
  matches: ['<all_urls>'],
  // 2. Set cssInjectionMode
  cssInjectionMode: 'ui',

  async main(ctx) {
    // 3. Define your UI
    const ui = await createShadowRootUi(ctx, {
      name: 'example-ui',
      position: 'inline',
      onMount: (container) => {
        // Render your app to the UI container
        const unmount = render(() => <App />, container);
      },
      onRemove: (unmount) => {
        // Unmount the app when the UI is removed
        unmount?.();
      },
    });

    // 4. Mount the UI
    ui.mount();
  },
});
```

**Comment by aklinker1** [comment]
Interesting... That definitely shouldn't be getting added for solid, I'll check this out soon. Thanks for the report.

**Comment by doompatrool** [comment]
> Interesting... That definitely shouldn't be getting added for solid, I'll check this out soon. Thanks for the report.

if you need more input i can provide. This error stopping me to keep testing wxt. Thanks

**Comment by aklinker1** [comment]
OK, so the problem here is that WXT uses esbuild and it's [`jsx: "automatic"](https://esbuild.github.io/content-types/#auto-import-for-jsx)` option when loading JSX entrypoints. 

I think I can switch to [this](https://esbuild.github.io/content-types/#using-jsx-without-react) instead, since we never actually execute the JSX while loading the options for the entrypoint during the build, we just need to avoid syntax errors.

**Comment by doompatrool** [comment]
> Yup, that got me past the build error:
> 
> https://github.com/wxt-dev/wxt/blob/c258b99cec82fea020e151b7f111c9ffa01d8c54/src/core/utils/building/import-entrypoint.ts#L103-L110
> 
> ```diff
> -    jsx: isJsx ? "automatic" : undefined,
> +   ...(isJsx ? { 
> +     "jsxFactory": "h",
> +     "jsxFragment": "Fragment"
> +   } : undefined),
> ```
> 
> Need to test with react, but I expect it will be fine.

Great job. thanks for working on fix so fast. I'll wait for the minor release with fix.

1. If i may suggest, some of us are not used to typescript and js normally is fine, so if not much of a big thing, a js/jsx starter template on install would be nice. 

2. Also i've noticed even after converting all files to js/jsx there is still tsconfig things inside `.wxt` folder, not sure if its doing anything but i saw on build that it keeps looking for .ts files.

**Comment by aklinker1** [comment]
> 1. If i may suggest, some of us are not used to typescript and js normally is fine, so if not much of a big thing, a js/jsx starter template on install would be nice.

That will double the maintenance I have to do around templates, so no. If anyone else wants to create and maintain community templates, I'm willing to integrate them into `wxt init`.

> 2. Also i've noticed even after converting all files to js/jsx there is still tsconfig things inside .wxt folder, not sure if its doing anything but i saw on build that it keeps looking for .ts files.

Yes, the time it takes to generate and write the types to that directory is negligible compared to the build itself. As for detecting TS files during the build, there's no option to disable that. If you want to use JS, go ahead. If you want to use TS, go ahead. Goal is to have zero overhead for both options, which I think the current setup supports quite well.

I also have about zero experience writing just JS, so I don't know what's preferable for a JS environment. `jsconfig.json` file or no? So I've kept generating the type information so you can do whatever you want.

**Comment by doompatrool** [comment]
> Released fix in [`v0.16.10`](https://github.com/wxt-dev/wxt/releases/tag/v0.16.10)

Sweet. Thank you!!
 
> I also have about zero experience writing just JS, so I don't know what's preferable for a JS environment. `jsconfig.json` file or no? So I've kept generating the type information so you can do whatever you want.

Got it. I don't have issue with the tsconfig files in general, now you mention all the support chores for js template makes sense.
